specutils/io/default_loaders/wcs_fits.py

```diff
@@ -96,6 +130,49 @@ def wcs1d_fits_loader(file_obj, spectral_axis_unit=None, flux_unit=None,
         else:
             data = u.Quantity(hdulist[hdu].data, unit=flux_unit)
 
+        # Read mask from specified HDU or try to auto-detect it.
+        mask = None
+        if mask_hdu is True:
+            for ext in ('MASK', 'DQ', 'QUALITY'):
+                if ext in hdulist and (isinstance(hdulist[ext], fits.ImageHDU) and
+                        hdulist[ext].data is not None):
+                    mask = hdulist[ext].data
+                    break
+        elif isinstance(mask_hdu, (int, str)):
```

Comment:
Why would mask be represented as `str`?

Reply:
`mask_hdu` is the generalised index to its HDU, which for a HDUList can be the numerical index or the label from `EXTNAME`.